### ISSUE TITLE
Eclipse/m2e-apt can't seem to handle code gen from src/main/java and src/test/java

### ISSUE THREAD

**Issue by alex-sherwin** [immutables/immutables#731]
When I've got a project with classes to be code-gen'd using `@Value.Immutable` under both `src/main/java` and `src/test/java`, Maven and IntelliJ have no problem with generating sources out to `target/generated-sources/annotations` and `target/generated-test-sources/test-annotations`

However, it seems that the Eclipse project config for the Java Compiler / Annotation Processing is limited to a single source/destination?

Hopefully I'm missing something...

**Comment by elucash** [comment]
yes, I found no way to separate those in Eclipse. I usually generate sources and build classes to a separate dirs in Eclipse so they will not clash with Maven's target (and don't include target/generate-sources... as a source folder in a project in this case). It's like having 2 systems: one is "official" command line build and the one in IDE is convenience/productivity tool, but doesn't replace an "official" one. For example, having _src_ and _test_ source folders in Eclipse doesn't make any difference wrt compile/test-compile dependency scopes etc, everything is shared within the module(project).

**Comment by vorburger** [comment]
> I usually generate sources and build classes to a separate dirs in Eclipse so they will not clash with Maven's target (and don't include target/generate-sources... as a source folder in a project in this case). It's like having 2 systems: one is "official" command line build and the one in IDE is convenience/productivity tool, but doesn't replace an "official" one.

FYI I also do this; it's too much of a PITA to mix them; so Eclipse goes to `target-ide` ([see here for how to](https://github.com/opendaylight/odlparent/blob/ace0bc1ea53ab0e9f86c0585d98585c3aa9da496/odlparent-lite/pom.xml#L351)).

> For example, having src and test source folders in Eclipse doesn't make any difference wrt compile/test-compile dependency scopes etc, everything is shared within the module(project).

FYI that is actually changing in upcoming Eclipse Oxygen; see https://bugs.eclipse.org/bugs/show_bug.cgi?id=224708 (and https://bugs.eclipse.org/bugs/show_bug.cgi?id=526858 for M2E)

> However, it seems that the Eclipse project config for the Java Compiler / Annotation Processing is limited to a single source/destination?

That should perhaps go as a proposed Enhancement to https://bugs.eclipse.org more than here? :smile:
